examples/stdlib/list/countIf.effekt

```diff
@@ -0,0 +1,29 @@
+module examples/pos/list/countIf
+
+import list
+
+def main() = {
+
+        val empty: List[String] = []
+
+        println(empty.countIf{ x => x.length() == 2}) // expects 0
+
+
+
+        val listStr_none: List[String] = ["A","B","C"]
+
+        println(listStr_none.countIf{ x => x.length() == 2}) // expects 0
+
+
+
+        val listStr_some: List[String] = [ "AB","BC","A","GG","AAA"]
+
+        println(listStr_some.countIf{ x => x.length() == 2}) // expects 3
+
+
+
+        val listInt : List[Int] =[1 , 2 , 3 , 4 , 5 , 6 , 7 , 8 , 9 , 10] // expects 5
+
+        println(listInt.countIf{x => (x* 2) > 10})
```

Comment:
In addition to the previously mentioned, please space out elements evenly and put the comments to the relevant parts.
```suggestion
  val numbers: List[Int] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
  println(numbers.countIf { x => (x * 2) > 10 })  // expects 5
```